packages/design-system/src/components/Autocomplete/Autocomplete.stories.tsx

```diff
@@ -139,6 +163,47 @@ export const LabeledList: Story = {
   } as any,
 };
 
+export const ItemGroups: Story = {
+  render: Template,
+  args: {
+    textFieldLabel: 'Select a state.',
+    textFieldHint:
+      'Type "A" then use ARROW keys to change options, ENTER key to make a selection, ESC to dismiss.',
+    items: [
+      makeGroup('Group A', [
+        makeItem('Alabama'),
+        makeItem('Alaska'),
+        makeItem('Arizona'),
+        makeItem('Arkansas'),
+      ]),
+      makeGroup('Group C', [makeItem('California'), makeItem('Colorado'), makeItem('Connecticut')]),
+      makeGroup('Group D', [makeItem('Delaware'), makeItem('District of Columbia')]),
+    ],
+  } as any,
+};
+
+export const MixedGroupsAndStandaloneItems: Story = {
+  render: Template,
+  args: {
+    textFieldLabel: 'Search for a healthcare specialty or doctor’s office.',
+    textFieldHint:
+      'Type to filter options. Use ARROW keys to navigate, ENTER to select, ESC to dismiss.',
+    items: [
+      makeItem('Care Clinic - Specialty Center'),
+      makeItem('Healthy Life Gastroenterology - Main Campus'),
+      makeItem('Dermatology Associates - East Wing'),
+      makeGroup('Healthcare Specialties', [
+        makeItem('Pediatrics'),
+        makeItem('Gastroenterology'),
+        makeItem('Dermatology'),
+        makeItem('Cardiology'),
+        makeItem('Neurology'),
+        makeItem('Orthopedics'),
+      ]),
+    ],
+  } as any,
```

Comment:
Does this have to be `any` because the type here is `Array< AutocompleteItem | AutocompleteItemGroup>` instead of `AutocompleteItem[] | AutocompleteItemGroup[]`?

Reply:
Would retyping `AutocompleteItems` to be an array of both item types be more accurate?

Reply:
> Does this have to be `any` because the type here is `Array<AutocompleteItem | AutocompleteItemGroup>` instead of `AutocompleteItem[] | AutocompleteItemGroup[]`?

@pwolfert, I think I might be missing something here, because it looks like the `as any` is applied to the `args` object, not directly to the `items` array. That said, I could probably apply more accurate type casting to the `items` array. Let me know if you had something else in mind!

Reply:
Oh, I didn't catch that. I thought it was for the array. I'm curious if you were to remove it if it would complain about the `items` array specifically along with any other errors it has.

Reply:
@tamara-corbalt instead of `as any`, you could do something like this `as AutocompleteProps & { textFieldLabel: string; textFieldHint: string }`